roles/lib_utils/src/class/yedit.py

```diff
@@ -512,112 +521,146 @@ def parse_value(inc_value, vtype=''):
         # we will convert to bool if it matches any of the above cases
         if isinstance(inc_value, str) and 'bool' in vtype:
             if inc_value not in true_bools and inc_value not in false_bools:
-                raise YeditException('Not a boolean type. str=[%s] vtype=[%s]'
-                                     % (inc_value, vtype))
+                raise YeditException('Not a boolean type. str=[{}] vtype=[{}]'.format(inc_value, vtype))
         elif isinstance(inc_value, bool) and 'str' in vtype:
             inc_value = str(inc_value)
 
+        # There is a special case where '' will turn into None after yaml loading it so skip
+        if isinstance(inc_value, str) and inc_value == '':
```

Comment:
@ashcrow, great question.  I would like to maybe visit it at some point as I find this a difficult topic.    The incoming value when passing _something_ to a module is generally defined in the module spec.  The problem here is that I want that value to be anything provided.  

The point of this module is to make a specific cut into a yaml file at a specific path and place a value.  Someone could pass in a dict, list, int, str, null, etc.

The problem comes in when loading the variable.  In ansible 1.9 it is different than ansible 2.x.  So I attempt to load the incoming value through yaml if it matches a certain data type.  If the incoming value is of type string and it is empty, the `yaml.safe_load` turns the `''` into `None`.  This is not the desired behavior.  We would like the string to show up in the yaml file as `''`.

I'm always open to discussions on how to handle incoming variable data that could be different types.  That would be helpful.